### ISSUE TITLE
Using BERTopic behind company firewall that blocks Internet access

### ISSUE THREAD

**Issue by pearlyquek** [MaartenGr/BERTopic#501]
Hi Maarten!

Would there be a way to download the transformation models so that I can point to them locally when running BERTopic? I'm behind a great IT firewall that doesn't allow me to access the Internet at all.

Thank you!

Cheers,
Pearly

**Comment by pearlyquek** [comment]
Thank you Maarten! Let me try this, and I will get back to you if it worked for me.

Meanwhile, can I seek your advice on the models I should prioritise downloading for my particular use case?

I work for a government agency that encourages multi-national corporations to invest and operate in my country. I have on hand about 60,000 call reports filed over 5 years. Call reports are formally written by folks in our agency to record-keep the key topics discussed during the meeting. Each call report can be a few paragraphs long. They are typically filed in full sentences, but may sometimes consist of bullet points as well. Typically, global business/technology trends are mentioned, as well as the company’s business/financial/tax strategies and investment considerations (e.g. availability of talent, resources, funding). I have been tasked to run a topic modelling analysis to discern trends over the past 5 years across different regions.

Thank you in advance for your help!

**Comment by MaartenGr** [comment]
Before going into the model that might work best for you, I would advise splitting up your documents into sentences or smaller paragraphs. Information is lost if the documents are too long since you are most likely to be working with sentence-transformers. 

There are a number of models that work quite well out-of-the-box, you can find them [here](https://www.sbert.net/docs/pretrained_models.html) but if you were to ask me, I would advise `all-mpnet-base-v2` or `paraphrase-multilingual-MiniLM-L12-v2` for multi-lingual documents. Both have shown great performance in my use cases. 

Having said that, it does not mean that they will always perform best. In practice, it might be worthwhile to go through some of the back-end possibilities and research which types of language models suite your use case best. You can find an overview of embedding models [here](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html).

**Comment by pearlyquek** [comment]
Thanks for your advice as always Maarten! Appreciate it greatly :)

**Comment by pearlyquek** [comment]
This worked! Thanks Maarten :)

**Comment by sankalpjain30** [comment]
Hi Maarten...Your embedding part works.....But what in case i want to use umap ? Its giving connection error .

# Initiate UMAP
umap_model = UMAP(n_neighbors=15, 
                  n_components=5, 
                  min_dist=0.0, 
                  metric='cosine', 
                  random_state=100)
# Initiate BERTopic
topic_model = BERTopic(umap_model=umap_model, language="english", calculate_probabilities=True)
# Run BERTopic model
topics, probabilities = topic_model.fit_transform(amz_review['review_lemmatized'])

**Comment by MaartenGr** [comment]
@sankalpjain30 UMAP should not be the culprit here as it does not require a connection. Could you share the exact error you are getting? Based on your code, I would actually think the culprit here would be the embedding stage.

**Comment by saikot-paul** [comment]
Hi there, for zero shot topic modelling you pass a string as the parameter if you were to use facebook, cohere, etc is it possible to load representational models the same way you can download an embedding model from sentence transformer?

**Comment by MaartenGr** [comment]
@saikot-paul That depends on whatever model you use. Many LLMs can be downloaded locally for you to use. Other models, like Cohere uses are APIs and are therefore not able to be used without an internet connection.
